LibTest/js_interop/FunctionToJSExportedDartFunction/toJSCaptureThis_A01_t03.dart

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion JSExportedDartFunction get toJSCaptureThis
+/// A callable JavaScript function that wraps this [Function] and captures the
+/// `this` value when called.
+///
+/// Identical to `toJS`, except the resulting [JSExportedDartFunction] will pass
+/// `this` from JavaScript as the first argument to the converted [Function].
+/// Any [Function] that is converted with this member should take in an extra
+/// parameter at the beginning of the parameter list to handle this.
+///
+/// @description Check that if this getter is called on a torn off constructor
+/// it is a run-time error to call it in JavaScript.
+/// @author sgrekhov22@gmail.com
+
+import 'dart:js_interop';
+import 'dart:js_interop_unsafe';
+import '../../../Utils/expect.dart';
+import '../js_utils.dart';
+
+@JS()
+extension type ET(JSNumber id) {
+  ET.c1(pThis, this.id) {
+    Expect.equals(globalContext, pThis);
+  }
+
+  @JS()
```

Comment:
Note that `@JS` does nothing on constructors here and below, regardless of `external` or not. It's still on my plate to add warnings though.